### PR TITLE
Extend readme and docs

### PR DESCRIPTION
**Pull request by tkrille**
This is some kind of draft and a request for comments, so that we get a better readme and docs in the main repository.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,21 +7,55 @@ open standards:
...
 
-OSIAM is published under the MIT licence, giving you the greatest freedom possible to
-utilize OSIAM in you project or product.
+OSIAM is published under the MIT licence, giving you the greatest freedom
+possible to utilize OSIAM in you project or product.
+
+## Use cases
+
+* Single app with OSIAM as free backing store for users and groups and local
+  auth service. [Learn more](docs/single-app-use-case.md).
```
Comment (wallner) [inline README.md]: I'm not a huge fan of using "click here" links. Why not turn the first two words into a link?

File: README.md [hunk 2]
```diff
@@ -7,21 +7,55 @@ open standards:
...
+* Distributed app with OSIAM as central identity store and auth service
+  [Learn more](docs/distributed-app-use-case.md).
+* Publish your app's API with OSIAM as auth service for 3rd-parties and allow
+  safe access to your user's data [Learn more](docs/protected-api-use-case.md).
```
Comment (wallner) [inline README.md]: Same here.

File: README.md [hunk 3]
```diff
@@ -7,21 +7,55 @@ open standards:
...
-* [auth-server](https://github.com/osiam/auth-server) - handles the authentication and authorization based on OAuth 2.0
-* [resource-server](https://github.com/osiam/resource-server) - holds the SCIM based user data
-* [addon-self-administration](https://github.com/osiam/addon-self-administration) - account self-service web application
-* [addon-administration](https://github.com/osiam/addon-administration) - administration web application for users and groups
+* [`auth-server`](https://github.com/osiam/auth-server)
+  handles the authentication and authorization based on OAuth 2.0
+* [`resource-server`](https://github.com/osiam/resource-server)
+  holds the SCIM based user data
+* [`addon-self-administration`](https://github.com/osiam/addon-self-administration)
+  provides account self-service as a web application
```
Comment (wallner) [inline README.md]: provides account management self-service as a web application

File: README.md (hunk 1 above)
Comment (dacrome) [inline README.md]: I think "click here" is a little bit different to this style. I like "Lean more". Everyone who is a fan of "Learn more" please "click here"!

File: README.md [hunk 4]
```diff
@@ -7,21 +7,55 @@ open standards:
...
+* [`addon-administration`](https://github.com/osiam/addon-administration)
+  lets you administer users and groups via a web application
+
+### Easy customization
+
+Every visual aspect of OSIAM's components can be easily customized by supplying
+your own templates, stylesheets and scripts to make OSIAM seamlessly integrate
```
Comment (wallner) [inline README.md]: integrate seamlessly (?)

File: README.md (hunk 2 above)
Comment (dacrome) [inline README.md]: Same here.

File: README.md [hunk 5]
```diff
@@ -7,21 +7,55 @@ open standards:
...
+with your application. Read the
+[customization guide](docs/customization-guide.md) to learn more.
 
 ## Issue Tracker
 
-Issues, bugs, and change and feature requests can be brought to us at [the issue tracker](https://github.com/osiam/osiam/issues).
+Issues, bugs, and change and feature requests can be brought to us at
```
Comment (wallner) [inline README.md]: Please report issues, bugs and feature requests via the issue tracker

File: docs/protected-api-use-case.md [hunk 6]
```diff
@@ -0,0 +1,4 @@
+# Protected APIs
+
+Do it like Twitter, Facebook or Google and expose your API to third party
```
Comment (wallner) [inline docs/protected-api-use-case.md]: I think this sentence should be rephrased.

**Comment by tkrille** [comment]
all done

**Comment by tkrille** [comment]
so, do we wanna merge this now or extend the documention first, i.e. fill all **TODO**s? It's definitely better than the current README/docs ... :) But my opinion on this might be biased ;)

**Pull request merged**